### PR TITLE
Move `/hardware/` under `/system/` as per RFD-288

### PR DESCRIPTION
**Pull request by zephraph**
Part of the work to #1675.

My current understanding as per my conversation with @karencfv last week is that `system` will be the root command in the CLI which these operator commands will be grouped under. As such, I've prepended `system` to the method name.

### CODE REVIEW

File: nexus/src/external_api/http_entrypoints.rs [hunk 1]
```diff
@@ -200,10 +200,10 @@ pub fn external_api() -> NexusApiDescription {
...
         api.register(vpc_firewall_rules_update)?;
 
-        api.register(rack_list)?;
-        api.register(rack_view)?;
-        api.register(sled_list)?;
-        api.register(sled_view)?;
+        api.register(system_rack_list)?;
+        api.register(system_rack_view)?;
+        api.register(system_sled_list)?;
+        api.register(system_sled_view)?;
```
Comment (zephraph) [inline nexus/src/external_api/http_entrypoints.rs:206]: @ahl I'd like your feedback on this specifically. Does the prefixing make sense to you or should we not? If we decide not to when I get to global images we might need to have a one off prefix for that resource to differentiate it.

**Comment by ahl** [comment]
> My current understanding as per my conversation with @karencfv last week is that `system` will be the root command in the CLI which these operator commands will be grouped under. As such, I've prepended `system` to the method name.

I think we should not prepend `system_` to all these methods, but I don't feel that strongly. I also think we should put them all under a `system` tag (we have too many tags right now, again IMO).

With regard to the CLI, our intention is to generate a big chunk of it... **and** to manually intervene. For example, we might take the `clap` hierarchy and then hoist all the subtrees with the `system` tag under a `system` subcommand.

All this is to say: we have several different consumers for the `operationId`: the CLI, the API, and the SDKs. I think in this case it makes more sense to do some manual intervention in the CLI than it does to impose a prefix for the API and SDKs, but I'd love to have that discussion.

**Comment by zephraph** [comment]
Sounds good! I'll remove it for this PR and update the tags.

**Comment by ahl** [comment]
thanks! there's tags review on my list. I should find a way to make these pending work items visible...

**Comment by karencfv** [comment]
I agree! `system_` does not have to be prepended. Having the endpoint path start with `/system/` should be enough imo :)

This will also make it clear for the terraform provider resource schema which is not (and most likely never be) generated based on tags or operation IDs.

**Comment by karencfv** [comment]
Thanks for working on this @zephraph btw!!!!

**Pull request merged**